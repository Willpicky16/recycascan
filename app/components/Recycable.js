import React, { Component } from "react";
import { StyleSheet, Text, View, AsyncStorage } from "react-native";
import axios from "axios";

const ROOT = 'https://world.openfoodfacts.org/api/v0/product/';
const BINROOT = 'https://vast-eyrie-43528.herokuapp.com/api/bins';

export default class Recycable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userDetails: {},
      product: {},
      bin: '',
      loading: true
    };
  }
  componentDidMount() {
    AsyncStorage.getItem('userDetails', (err, result) => {
      let val = JSON.parse(result);
      this.setState({
        userDetails: val
      });
    });
    axios
      .get(`${ROOT}/${this.props.navigation.state.params.code}`)
      .then(res => {
        axios
          .get(`${BINROOT}?packaging=${res.data.product.packaging}&council=${this.state.userDetails.council}`)
          .then((res) => {
            this.setState({
              bin: res.data.bins[0].bin,
              loading: false
            })
            console.log(this.state);
          })
          .catch((err) => {
            console.log(err);
          })
        this.setState({
          product: res.data.product
        });
      })
      .catch(err => {
        console.log(err);
        this.props.navigation.navigate('ProductSubmit', {code: this.props.navigation.state.params.code});
      });
  }
  render() {
    if (this.state.loading === true)
      return (
        <View style={styles.container}>
          <Text style={styles.loading}>loading....</Text>
        </View>
      );
    return (
      <View style={styles.container}>
        <Text style={styles.subtitle}>{this.state.product.product_name} ({this.state.product.code})</Text>
        <Text style={styles.subtitle}>
          Put in {this.state.bin} bin!
        </Text>
      </View>
    );
  }
}

// function getBinColour () {
//   axios
//     .get(`${BINROOT}?packaging=${res.data.product.packaging}&council=${this.state.council}`)
//     .then((res) => {
//       console.log(res);
//     })
//     .catch((err) => {
//       console.log(err);
//     })
// }

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  title: {
    color: "blue",
    fontSize: 25
  },
  subtitle: {
    fontSize: 20
  },
  loading: {
    color: "red",
    fontSize: 30
  }
});
