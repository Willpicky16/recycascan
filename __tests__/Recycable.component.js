import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Recycable from '../app/components/Recycable';

test('recycable page renders correctly', () => {
  const tree = renderer.create(
    <Recycable />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
