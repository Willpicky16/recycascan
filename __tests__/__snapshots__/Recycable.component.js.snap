// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recycable page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Recycable
  </Text>
</View>
`;
